controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -635,17 +651,27 @@ private void closeConnection(Reply reply, RawClient client) {
     }
 
     private <T extends Request & WireCommand> CompletableFuture<Reply> sendRequest(RawClient connection, long requestId, T request) {
-        return connection.sendRequest(requestId, request)
+        CompletableFuture<Reply> future = Futures.futureWithTimeout(
+                () -> connection.sendRequest(requestId, request)
+                                .exceptionally(e -> {
```

Comment:
nit: Can we have all the exception handling in one block? it will reduce the creation of a CompletionStage and all the error handling will be done in one block (e.g: right before returning the future below)